server/WebAPI/Services/UserService.cs

```diff
@@ -1,45 +1,70 @@
 using Entities;
 using Entities.Models;
-using System;
-using System.Security.Cryptography;
-using System.Text;
 
 namespace WebAPI.Services
 {
-    public class UserService
+  public class UserService
+  {
+
+    private readonly UserRepository _userRepository;
+
+    public UserService(UserRepository userRepository)
     {
+      _userRepository = userRepository;
+    }
 
-        private readonly UserRepository _userRepository;
-        
-        public UserService(UserRepository userRepository)
-        {
-            _userRepository = userRepository;
-        }
+    public User GetUser(string email, string password)
+    {
+      var user = _userRepository.FindUserByEmail(email);
+      if(user != null && CryptoUtils.VerifyPassword(password, user.Password))
+      {
+        return user;
+      }
+      return null;
+    }
 
-        public User GetUser(string email, string password)
-        {
-            var user = _userRepository.FindUserByEmail(email);
-            if (user != null && CryptoUtils.VerifyPassword(password, user.Password))
-            {
-                return user;
-            }
-            return null;
-        }
+    public bool IsUserExist(string email)
+    {
+      return _userRepository.FindUserByEmail(email) != null;
+    }
+
+    public void CreateUser(string email, string password)
+    {
+      var user = new User
+      {
+        Email = email,
+        Password = CryptoUtils.HashPassword(password)
+      };
+
+      _userRepository.CreateUser(user);
+    }
+    public User GetUser(string email)
+    {
+      return _userRepository.FindUserByEmail(email);
```

Comment:
will this return User data including password?

Reply:
Yes, it will return user data including the hashed password. a token is created based on the user. I had no desire to rewrite the token creation with minor changes